librz/analysis/var.c

```diff
@@ -46,7 +46,17 @@ RZ_API bool rz_analysis_function_rebase_vars(RzAnalysis *a, RzAnalysisFunction *
 // If the type of var is a struct,
 // remove all other vars that are overlapped by var and are at the offset of one of its struct members
 static void shadow_var_struct_members(RzAnalysisVar *var) {
-	RzBaseType *btype = rz_type_db_get_base_type(var->fcn->analysis->typedb, var->type);
+	if (var->type->kind == RZ_TYPE_KIND_POINTER || var->type->kind == RZ_TYPE_KIND_CALLABLE) {
+		// Pointers don't shadow anything
+		return;
+	}
+	if (var->type->kind == RZ_TYPE_KIND_ARRAY) {
+		// TODO: support arrays
+		return;
+	}
+	// In other cases we check the type by name
+	const char *tname = var->type->identifier.name;
+	RzBaseType *btype = rz_type_db_get_base_type(var->fcn->analysis->typedb, tname);
```

Comment:
I think what we actually want to do is calculate the size of `var->type` (in its own API) and then just radically shadow all overlaps. Doesn't have to be in this pr, but maybe add a comment.